### PR TITLE
mat-table-filter bugfix and enhancement

### PR DESCRIPTION
**Pull request by dtarjeft**
[Primitive array support](https://github.com/HalitTalha/mat-table-extensions/issues/13) as well as [handling 'false'](https://github.com/HalitTalha/mat-table-extensions/issues/12)

### CODE REVIEW

File: projects/mat-table-filter/src/lib/mat-table-filter.service.ts [hunk 1]
```diff
@@ -13,12 +13,26 @@ export class MatTableFilterService {
...
+        if(exampleColumnValue.length === 0 || typeof exampleColumnValue[0] === 'object' || typeof exampleColumnValue[0] === 'function' 
+          || !itemColumnValue || itemColumnValue.length === 0){
+          continue; // undefined/null/empty arrays mean the caller wants all the data. Also, skip non-primitives.
+        }
+        for (let j = 0; j < exampleColumnValue.length; j++) {
+          if (!exampleColumnValue[j]) {
+            continue;
+          }
+          if(itemColumnValue.indexOf(exampleColumnValue[j]) === -1){
+            return false;
```
Comment (dtarjeft) [inline projects/mat-table-filter/src/lib/mat-table-filter.service.ts]: Could perhaps add functionality to switch between `OR` vs `AND` here at some point- I only needed `AND` for my use case, but I could see `OR` coming in handy.

**Comment by HalitTalha** [comment]
Many thanks for your help.
I'm really sorry to be late about this. I've actually implemented a solution with loadash providing some more features, I just couldn't find the time to publish and commit it. That's why I have to close this pull request this time. I'd be appriciated if you keep up the good work :)

File: projects/mat-table-filter/src/lib/mat-table-filter.service.ts [hunk 2]
```diff
@@ -13,12 +13,26 @@ export class MatTableFilterService {
       const exampleColumn = exampleEntityObjectKeys[i];
       const exampleColumnValue = exampleEntity[exampleColumn];
       const itemColumnValue = item[exampleColumn];
-      if (!exampleColumnValue || exampleColumnValue instanceof Array) {
+      if (!exampleColumnValue && typeof exampleColumnValue !== 'boolean') {
```
Comment (HalitTalha) [inline projects/mat-table-filter/src/lib/mat-table-filter.service.ts]: I'll approve this part and I have to override the rest I'm afraid. Thanks again.

Comment (dtarjeft) [reply projects/mat-table-filter/src/lib/mat-table-filter.service.ts]: If you're taking the boolean fix you'll want [this line](https://github.com/HalitTalha/mat-table-extensions/pull/16/files#diff-3d4f8d536352779b2c0ccd3f9ad818b0R35) too. Glad you found a better implementation with lodash! I'm looking forward to seeing those features 👍

**Comment by HalitTalha** [comment]
Thanks again for your effort. Appearantly I have to override it all since Ive gone for some more changes along with other enhancements. Thats the reason Im closing this PR. Ive committed the new features in ng6-7 branch they will be available for master soon.

**Pull request closed without merging**